airbyte-cdk/python/airbyte_cdk/sources/declarative/extractors/jq.py

```diff
@@ -24,7 +25,23 @@ def __init__(self, transform: str, decoder: Decoder, config, kwargs=None):
         self._kwargs = kwargs
         self._decoder = decoder
 
-    def extract_records(self, response: requests.Response) -> List[Record]:
+        if record_filter:
```

Comment:
I think we should move this filter to another extractor class which would
1. delegate to another extractor
2. apply the filter

This would allow us to use the same filtering logic with any type of record extractor. This will be handy once we start having more complex transforms that can't be easily expressed in a single jq query (eg for adding, removing, or renaming fields)

Reply:
interesting, i do like how flexible this would be. I'll add in a top level `record_extractor` that will allow for the first two but adding in a new step for transform should be relatively straightforward when we need it. The language will look something like this:
```
record_extractor:
  fetcher:
    class_name: airbyte_cdk.sources.declarative.extractors.jq.JqFetcher
    decoder: "*ref(decoder)"
  filter:
    class_name: airbyte_cdk.sources.declarative.extractors.ConditionalFilter
    condition: "{{ record['updated_at'] > stream_state['obj']['updated_at'] }}"
  transformer:
    class_name: airbyte_cdk.sources.declarative.extractors.FieldTranformer
    transform: 
```

Somewhat related, but what do you think about renaming `extractor` to `record_extractor`? Its a bit verbose, but I was originally appending record to each field (i.e. `record_fetcher`, `record_filter`, `record_transformer`) since it felt unclear what the subcomponents were doing.

Reply:
`record_selector` is fine with me